Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (C) 2024 Igalia S.L.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Library General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Library General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Library General Public License
+ *  along with this library; see the file COPYING.LIB.  If not, write to
+ *  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include <wtf/StdLibExtras.h>
+#include <wtf/glib/GRefPtr.h>
+
+namespace WTF {
+
+inline std::span<const uint8_t> span(GBytes* bytes)
+{
+    size_t size = 0;
+    const auto* ptr = static_cast<const uint8_t*>(g_bytes_get_data(bytes, &size));
+    return unsafeForgeSpan<const uint8_t>(ptr, size);
+}
+
+inline std::span<const uint8_t> span(const GRefPtr<GBytes>& bytes)
+{
+    return span(bytes.get());
+}
```

Comment:
Hm, in retrospect, I probably would have not added this overload. All it does is save you from writing `.get()` which doesn't seem like a great reason to have twice as many functions here.

Reply:
I guess this is a pretty minor complaint, but it does add up when adding additional helper functions here, e.g. for https://bugs.webkit.org/show_bug.cgi?id=282239 I currently have:

```
diff --git a/Source/WTF/wtf/glib/GSpanExtras.h b/Source/WTF/wtf/glib/GSpanExtras.h
index 8c7c9d57e23d..623eaeba926c 100644
--- a/Source/WTF/wtf/glib/GSpanExtras.h
+++ b/Source/WTF/wtf/glib/GSpanExtras.h
@@ -36,6 +36,28 @@ inline std::span<const uint8_t> span(const GRefPtr<GBytes>& bytes)
     return span(bytes.get());
 }
 
+inline std::span<const uint8_t> span(GVariant* variant)
+{
+    const auto* ptr = static_cast<const uint8_t*>(g_variant_get_data(variant));
+    size_t size = g_variant_get_size(variant);
+    return unsafeForgeSpan(const uint8_t>(ptr, size));
+}
+
+inline std::span<const uint8_t> span(const GRefPtr<GVariant>& variant)
+{
+    return span(variant.get());
+}
+
+inline std::span<const uint8_t> span(GByteArray* array)
+{
+    return unsafeForgeSpan<const uint8_t>(array->data, array->len);
+}
+
+inline std::span<const uint8_t> span(const GRefPtr<GByteArray>& array)
+{
+    return span(array.get());
+}
+
 } // namespace WTF
 
 using WTF::span;
 ```

Reply:
TBH, I would rather write two helper functions one single time, than using `.get()` all over the code base without need. Changing all the:

```cpp
auto data = span(bytes);
```

to:

```cpp
auto data = span(bytes.get());
```

would only add noise, and require more typing each time one of the `span()` helper functions are used.

Reply:
The other option is to add an implicit conversion to allow a `GRefPtr<T>` to decay into `T*`, but I am not sure that's a good idea, given that writing a one-line overload while keeping explicit what is happening&mdash;I do not enjoy much the idea of implicit conversions.

Reply:
Let's not allow implicit conversion. I'll just keep things as-is.